deepchecks/vision/dataset.py

```diff
@@ -179,3 +188,81 @@ def validate_dataset(cls, obj) -> 'VisionDataset':
             raise DeepchecksValueError('Check requires a non-empty dataset')
 
         return obj
+
```

Comment:
Add a public function to call the _sample_data generator.

Reply:
why? where else it will be used?

Reply:
Left this before I saw you added caching. Thought this will be called from outside